### PR TITLE
ID-1051 [Feat] Support for starting a conversation by providing a contact's email via a query parameter

### PR DESCRIPTION
**Pull request by squallstar**
**Using the `contactEmail` parameter with a user’s email, start or open a conversation.** 

---

### Acceptance criteria:

1. I want to be able to start or open a conversation using contactEmail parameter
2. I want to be able to be told that I’m using the wrong Data source when the user wat not found for the given email

```
?contactEmail=john@example.org
```

### CODE REVIEW

File: js/build.js [hunk 1]
```diff
@@ -2756,6 +2768,7 @@ Fliplet.Widget.instance('chat', function (data) {
           Fliplet.UI.Toast.dismiss();
         }
       }).catch(function (err) {
+        console.error(err);
```
Comment (tonytlwu) [inline js/build.js:2771]:
```suggestion
        console.error(err);

```

File: js/build.js [hunk 2]
```diff
@@ -2743,6 +2740,21 @@ Fliplet.Widget.instance('chat', function (data) {
       $wrapper.removeClass('error');
 
       getContacts(false).then(function() {
+        if (userEmail) {
+          const user = _.find(otherPeople, function (user) {
+            return user.data[data.primaryKey] === userEmail;
+          });
+
+          if (!user) {
+            Fliplet.UI.Toast.error('The user has not been found. Please check whether the contacts data source has been set up correctly.');
```
Comment (tonytlwu) [inline js/build.js:2749]:
```suggestion
            Fliplet.UI.Toast.error('User is not found. Please check the contacts data source.');
```

File: js/build.js [hunk 3]
```diff
@@ -2743,6 +2740,21 @@ Fliplet.Widget.instance('chat', function (data) {
       $wrapper.removeClass('error');
 
       getContacts(false).then(function() {
+        if (userEmail) {
+          const user = _.find(otherPeople, function (user) {
```
Comment (tonytlwu) [inline js/build.js:2744]:
```suggestion
          var user = _.find(otherPeople, function (user) {
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/build.js (hunk 3 above)
Comment (squallstar) [reply js/build.js:2744]: woops!

**Review by tonytlwu — APPROVED** [review]

**Pull request merged**